src/cls/refcount/cls_refcount_ops.h

```diff
@@ -125,5 +124,149 @@ struct cls_refcount_read_ret {
 };
 WRITE_CLASS_ENCODER(cls_refcount_read_ret)
 
+struct chunk_refcount {
```

Comment:
i would drop this type and just use hobject_t.  i don't think we need to know which part of the referring object is pointing at this chunk, and we're missing the hash value here, which would be useful for a space-efficient coarse-grained ref map.